### PR TITLE
Support additional event hub consumer groups for testing environments

### PR DESCRIPTION
**Pull request by and-rewsmith**

### CODE REVIEW

**Comment by and-rewsmith** [comment]
I opened this PR so that I could test on one agent in isolation. This should not be merged until long haul for 1.0.8.3 has wrapped up.

File: builds/e2e/templates/longhaul-deploy-windows.yaml [hunk 1]
```diff
@@ -34,5 +34,5 @@ steps:
...
         $testName="LongHaul"
         $ArtifactInfo=convertfrom-stringdata (get-content $(Agent.HomeDirectory)/../artifacts/${{ parameters['images.artifact.name'] }}/artifactInfo.txt -raw)
-        $(Agent.HomeDirectory)/../artifacts/${{ parameters['images.artifact.name'] }}/scripts/windows/test/Run-E2ETest.ps1 -E2ETestFolder "$(Agent.HomeDirectory)/.." -ReleaseLabel "${{ parameters['release.label'] }}" -ArtifactImageBuildNumber "$($ArtifactInfo.BuildNumber)" -TestName "$testName" -ContainerRegistry "${{ parameters['container.registry'] }}" -ContainerRegistryUsername "${{ parameters['container.registry.username'] }}" -ContainerRegistryPassword "${{ parameters['container.registry.password'] }}" -IoTHubConnectionString "${{ parameters['iotHub.connectionString'] }}" -EventHubConnectionString "${{ parameters['eventHub.connectionString'] }}" -SnitchBuildNumber "${{ parameters['snitch.build.number'] }}" -SnitchStorageAccount "${{ parameters['snitch.storage.account'] }}" -SnitchStorageMasterKey "${{ parameters['snitch.storage.masterKey'] }}" -SnitchAlertUrl "${{ parameters['snitch.alert.url'] }}" -LoadGenMessageFrequency "${{ parameters['loadGen.message.frequency'] }}"
```
Comment (philipktlin) [inline builds/e2e/templates/longhaul-deploy-windows.yaml]: missing analyzer.consumerGroupId: '' at line 15.

Comment (and-rewsmith) [reply builds/e2e/templates/longhaul-deploy-windows.yaml]: Oops, good catch. I must have checked stress test deploy yaml when I was double-checking. 

Successful deployment here: 
https://msazure.visualstudio.com/One/_build/results?buildId=25895323

Comment (and-rewsmith) [reply builds/e2e/templates/longhaul-deploy-windows.yaml]: I SSH'd into each agent running long haul, except for AMD64 which has the wrong key in keyvault. They all report correct consumer group ids in the analyzer logs.

File: scripts/linux/runE2ETest.sh [hunk 2]
```diff
@@ -170,7 +170,7 @@ function prepare_test_from_artifacts() {
                 local escapedSnitchAlertUrl
                 local escapedBuildId
                 sed -i -e "s@<Analyzer.EventHubConnectionString>@$EVENTHUB_CONNECTION_STRING@g" "$deployment_working_file"
-                sed -i -e "s@<Analyzer.ConsumerGroupId>@${EVENT_HUB_CONSUMER_GROUP:-\$Default}@g" "$deployment_working_file"
+                sed -i -e "s@<Analyzer.ConsumerGroupId>@${EVENT_HUB_CONSUMER_GROUP_ID:-\$Default}@g" "$deployment_working_file"
```
Comment (philipktlin) [inline scripts/linux/runE2ETest.sh]: Move the default value after line 972

File: scripts/windows/test/Run-E2ETest.ps1 [hunk 3]
```diff
@@ -44,6 +44,9 @@
     .PARAMETER EventHubConnectionString
         Event hub connection string for receive D2C messages
 
+    .PARAMETER EventHubConsumerGroupId
+        Event hub consumer group id needed to avoid overloading the default Event Hub endpoint
```
Comment (philipktlin) [inline scripts/windows/test/Run-E2ETest.ps1]:
```suggestion
        Event hub consumer group id used by Analyzer module to subscribe events from Event Hub endpoint
```

File: scripts/windows/test/Run-E2ETest.ps1 [hunk 4]
```diff
@@ -206,6 +209,8 @@ Param (
     [ValidateNotNullOrEmpty()]
     [string] $EventHubConnectionString = $(Throw "Event hub connection string is required"),
 
+    [string] $EventHubConsumerGroupId = $null,
```
Comment (philipktlin) [inline scripts/windows/test/Run-E2ETest.ps1]: you need to provide default value here.

**Pull request merged**